### DISCUSSION TITLE
How to change a line of tags to a list of tags in yaml?

### DISCUSSION THREAD

**Discussion by francisHugo** [SilentVoid13/Templater#1347] [Help] [answered]
I have already had a bunch of notes with tags in yaml in a line, such as 
```
---
tags: tag1 tag2 tag3 ...
---
```

what I would like to do, whether can change these tags in to a list, like this:

```
---
tags:
  - tag1
  - tag2
  - tag3
  - ...
---
```

may a template do this?

I had try the linter plugin, with several possible settings I guessed, none of them works. So, I think maybe template can handle it.

thanks for helps.

**Comment by Zachatoo** [comment, marked as answer]
Something like this should work

```js
<%*
const file = tp.file.find_tfile(tp.file.path(true));
await app.fileManager.processFrontMatter(file, (frontmatter) => {
  frontmatter.tags = frontmatter.tags.split(" ");
});
-%>
```

If you're interested in running this on all your notes at once, you may be interested in this script
https://zachyoung.dev/posts/templater-snippets#run-obsidian-frontmatter-formatter-on-every-file

**Comment by cdongt** [comment]
1. You can use a  plugin:  Linter, which can easily implement this functionality.
2. Or  convert multi-line input to a list:
```
---
Tags:
<%*
let multiLineInput = await tp.system.prompt("Please input the tags", "", true, true);
let lines = multiLineInput.split("\n"); // Split multi-line text into an array.
lines.forEach(line => {
    if (line.trim() !== "") {
        tR += `- ${line}\n`; // Each line is preceded by a list symbol.
    }
});
%>
---
```
Here's how it works when you create a new file using this template:
![image](https://github.com/user-attachments/assets/b3e3d027-bd72-4bc9-9730-dd05ec3efeb4)
Reading View:
![image](https://github.com/user-attachments/assets/9bba2ffc-0560-453f-83d8-652f2a8884f2)
Source Code Mode:
![image](https://github.com/user-attachments/assets/56518343-4dad-4536-b211-6b3bd45547f7)
